apps/organisations/mixins.py

```diff
@@ -0,0 +1,76 @@
+from django import forms
+from django.urls import reverse
+from django.utils.html import mark_safe
+from django.utils.translation import gettext_lazy as _
+
+from apps.organisations.models import OrganisationTermsOfUse
+
+
+class OrganisationTermsOfUseMixin(forms.ModelForm):
+    organisation_terms_of_use = forms.BooleanField(
+        required=False,
+        help_text=_(
+            'You can still manage all your preferences on User Agreements.'
+        )
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user')
+        super().__init__(*args, **kwargs)
+
+        self.fields['organisation_terms_of_use'].label = \
+            self._get_terms_of_use_label
+
+        if OrganisationTermsOfUse.objects.filter(
+            user=self.user,
+            organisation=self.module.project.organisation
+        ).exists() and OrganisationTermsOfUse.objects.get(
+            user=self.user,
+            organisation=self.module.project.organisation
+        ).has_agreed is True:
```

Comment:
Wouldn't it be the same to just check if the entry with has_agreed=True exists? Then it would only be one query.. So I mean
`if OrganisationTermsOfUse.objects.filter(
            user=self.user,
            organisation=self.module.project.organisation,
            has_agreed=True
        ).exists()`

Reply:
And maybe it could go into the model as a property or method? Sth like user_has_agreed(user, organisation)? But not sure if that makes sense.. just thought we will probably need this again at other places.

Reply:
Yes, both good ideas! Which one should I do now? :)
Will start with trying the second version.

Reply:
I put it into the user model, because that is where we also should put the list for the overview.

Reply:
Perfect!